sdk/src/asset_handlers/gif_io.rs

```diff
@@ -0,0 +1,1453 @@
+        assert_eq!(
+            blocks.get(3),
+            Some(&BlockMarker {
+                start: 781,
+                len: 15,
+                block: test_block.clone()
+            })
+        );
+        assert_eq!(
+            blocks.get(4),
+            Some(&BlockMarker {
+                start: 796,
+                len: 15,
+                block: test_block
+            })
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_write_bytes() -> Result<()> {
+        let mut stream = Cursor::new(SAMPLE1);
+
+        let gif_io = GifIO {};
+
+        assert!(matches!(
+            gif_io.read_cai(&mut stream),
+            Err(Error::JumbfNotFound)
+        ));
+
+        let mut output_stream = Cursor::new(Vec::with_capacity(SAMPLE1.len() + 7));
+        let random_bytes = [1, 2, 3, 4, 3, 2, 1];
+        gif_io.write_cai(&mut stream, &mut output_stream, &random_bytes)?;
+
+        let data_written = gif_io.read_cai(&mut output_stream)?;
+        assert_eq!(data_written, random_bytes);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_write_bytes_at_existing_block() -> Result<()> {
+        let mut stream = Cursor::new(SAMPLE1);
+
+        let gif_io = GifIO {};
+
+        assert!(matches!(
+            gif_io.read_cai(&mut stream),
+            Err(Error::JumbfNotFound)
+        ));
+
+        let mut output_stream1 = Cursor::new(Vec::with_capacity(SAMPLE1.len() + 7));
+        let random_bytes = [1, 2, 3, 4, 3, 2, 1];
+        gif_io.write_cai(&mut stream, &mut output_stream1, &random_bytes)?;
+
+        let data_written = gif_io.read_cai(&mut output_stream1)?;
+        assert_eq!(data_written, random_bytes);
+
+        let mut output_stream2 = Cursor::new(Vec::with_capacity(SAMPLE1.len() + 3));
+        let random_bytes = [1, 2, 1];
+        gif_io.write_cai(&mut output_stream1, &mut output_stream2, &random_bytes)?;
+
+        let data_written = gif_io.read_cai(&mut output_stream2)?;
+        assert_eq!(data_written, random_bytes);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_write_bytes_replace() -> Result<()> {
+        let mut stream = Cursor::new(SAMPLE1);
+
+        let gif_io = GifIO {};
+
+        assert!(matches!(
+            gif_io.read_cai(&mut stream),
+            Err(Error::JumbfNotFound)
+        ));
+
+        let mut output_stream1 = Cursor::new(Vec::with_capacity(SAMPLE1.len() + 7));
+        let random_bytes = [1, 2, 3, 4, 3, 2, 1];
+        gif_io.write_cai(&mut stream, &mut output_stream1, &random_bytes)?;
+
+        let data_written = gif_io.read_cai(&mut output_stream1)?;
+        assert_eq!(data_written, random_bytes);
+
+        let mut output_stream2 = Cursor::new(Vec::with_capacity(SAMPLE1.len() + 3));
+        let random_bytes = [3, 2, 1, 2, 3];
+        gif_io.write_cai(&mut output_stream1, &mut output_stream2, &random_bytes)?;
+
+        let data_written = gif_io.read_cai(&mut output_stream2)?;
+        assert_eq!(data_written, random_bytes);
+
```

Comment:
An interesting sanity check here would be to see if "stream" matches SAMPLE1.  The contents should be identical after these transformation.